### ISSUE TITLE
Wrong semantics of binding equation for parameters with fixed=false in MSL

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#923]
**Reported by fcasella on 5 Dec 2012 18:32 UTC**
Model Modelica.Fluid.Valves.BaseClasses.PartialValve contains the following lines:

```
parameter SI.Area Av(
  fixed=if CvData == Modelica.Fluid.Types.CvTypes.Av then true
    else false,
  start=m_flow_nominal/(sqrt(rho_nominal*dp_nominal))*
        valveCharacteristic(opening_nominal)) = 0
   "Av (metric) flow coefficient";

...

initial equation
  if CvData == CvTypes.Kv then
    Av = Kv*Kv2Av "Unit conversion";
  elseif CvData == CvTypes.Cv then
    Av = Cv*Cv2Av "Unit conversion";
  end if;
```

The parameter Av has a binding equation, but it can have fixed = false depending on the value of the CvData enum parameter. The rationale is that one might provide set CvData to CvTypes.Av and provide the Av parameter directly via a modifier, or set CvData to any other value, in which case the parameter value is determined by a suitable initial equation.

The code currently present in MSL 3.2.1 (and previous versions) has a default binding equation = 0 in the parameter declaration, which remains in place in all cases when the Av parameters is not supplied explicitly by a modifier.

This is actually against the Modelica specification, which states in section 8.6 that ""All variables declared as parameter having fixed = false are treated as unknowns during the initialization phase, i.e. there must be additional equations for them – and the start-value can be used as a guess-value during initialization.
[In the case a parameter has both a binding equation and fixed = false a diagnostics is recommended, but the parameter should be solved from the binding equation.] " 

If CvData is not equal to CvTypes.Av, the binding equation Av = 0 is still there, and it obviously conflicts with the initial equation, generating an overconstrained inconsistent initialization system. For some reason, when fixed = false Dymola actually ignores the binding equation, so this problem was not found in the past, but behaviour is not conforming to the language specification.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/923

**Comment by modelica-trac-importer** [comment]
**Comment by fcasella on 7 Dec 2012 15:04 UTC**
Fixed in 5ba8fbbfacaf15bb75fedda45525fe98602cf7fa

**Comment by modelica-trac-importer** [comment]
**Comment by pharman on 9 Dec 2012 10:20 UTC**
The same issue occurs elsewhere in the MSL.

In Modelica.Mechanics.MultiBody.Examples.Elementary.InitSpringConstant there is the following declaration:
```
Modelica.Mechanics.MultiBody.Forces.Spring spring(s_unstretched=0.1,
    c(fixed=false) = 100);
```
This should presumably be:
```
Modelica.Mechanics.MultiBody.Forces.Spring spring(s_unstretched=0.1,
    c(fixed=false, start=100));
```

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 9 Dec 2012 12:59 UTC**
Replying to [comment:2 pharman]:


> The same issue occurs elsewhere in the MSL.
> 
> In Modelica.Mechanics.[wiki:MultiBody].Examples.Elementary.[wiki:InitSpringConstant] there is the following declaration:
> ```
> Modelica.Mechanics.[wiki:MultiBody].Forces.Spring spring(s_unstretched=0.1,
> c(fixed=false) = 100);
> ```
> This should presumably be:
> {{{
> Modelica.Mechanics.[wiki:MultiBody].Forces.Spring spring(s_unstretched=0.1,
> c(fixed=false, start=100));}}}

You are right, this is not correct Modelica.

Fixed in 6fe10f302f6f316d7265bfb2ce31366ce44eff04

**Comment by modelica-trac-importer** [comment]
**Modified by otter on 9 Dec 2012 13:00 UTC**
